client/src/favorites-provider.tsx

```diff
@@ -0,0 +1,68 @@
+import _ from 'lodash';
```

Comment:
how about adding this to the `context` folder? :)

Reply:
or might be better to have `hooks` folder and expose provider and `useFavourite` hook which would use context internally

Reply:
what about `providers` folder?

Reply:
we can have both of them 😅

Reply:
i'd like to keep pieces of logic like this in one place.. `contexts` and `hooks` sound really specific. `providers` on the other hand provide both at the same time.

Reply:
makes sense, we can go for it for now